File: Example/RecordLife/Classes/View/PositiveNegativeBarChartCell.m
//
//  HomeSixTableViewCell.m
//  RecordLife
//
//  Created by 谢俊逸 on 17/04/2017.
//  Copyright © 2017 谢俊逸. All rights reserved.
//

#import "PositiveNegativeBarChartCell.h"
#import "XJYChart.h"

@implementation PositiveNegativeBarChartCell

- (void)awakeFromNib {
    [super awakeFromNib];
    // Initialization code
}

- (void)setSelected:(BOOL)selected animated:(BOOL)animated {
    [super setSelected:selected animated:animated];
    
    // Configure the view for the selected state
}

- (instancetype)initWithStyle:(UITableViewCellStyle)style reuseIdentifier:(NSString *)reuseIdentifier {
    if (self = [super initWithStyle:style reuseIdentifier:reuseIdentifier]) {
        
        NSMutableArray *itemArray = [[NSMutableArray alloc] init];
        XJYBarItem *item1 = [[XJYBarItem alloc] initWithDataNumber:@(-80.93) color:XJYDarkBlue dataDescribe:@"test"];
        [itemArray addObject:item1];
        XJYBarItem *item2 = [[XJYBarItem alloc] initWithDataNumber:@(-107.04) color:XJYDarkBlue dataDescribe:@"test"];
        [itemArray addObject:item2];
        XJYBarItem *item3 = [[XJYBarItem alloc] initWithDataNumber:@(77.99) color:XJYDarkBlue  dataDescribe:@"test"];
        [itemArray addObject:item3];
        XJYBarItem *item4 = [[XJYBarItem alloc] initWithDataNumber:@(57.48) color:XJYDarkBlue dataDescribe:@"test"];
        [itemArray addObject:item4];
        XJYBarItem *item5 = [[XJYBarItem alloc] initWithDataNumber:@(-89.91) color:XJYDarkBlue dataDescribe:@"test"];
        [itemArray addObject:item5];
        
        XJYBarItem *item6 = [[XJYBarItem alloc] initWithDataNumber:@(66.93) color:XJYDarkBlue dataDescribe:@"test"];
        [itemArray addObject:item6];
        XJYBarItem *item7 = [[XJYBarItem alloc] initWithDataNumber:@(7.04) color:XJYDarkBlue dataDescribe:@"test"];
        [itemArray addObject:item7];
        XJYBarItem *item8 = [[XJYBarItem alloc] initWithDataNumber:@(-77.99) color:XJYDarkBlue  dataDescribe:@"test"];
        [itemArray addObject:item8];
        XJYBarItem *item9 = [[XJYBarItem alloc] initWithDataNumber:@(-28.48) color:XJYDarkBlue dataDescribe:@"test"];
        [itemArray addObject:item9];
        XJYBarItem *item10 = [[XJYBarItem alloc] initWithDataNumber:@(52.91) color:XJYDarkBlue dataDescribe:@"test"];
        [itemArray addObject:item10];
        //
        XJYBarItem *item11 = [[XJYBarItem alloc] initWithDataNumber:@(-0.93) color:XJYDarkBlue dataDescribe:@"test"];
        [itemArray addObject:item11];
        XJYBarItem *item12 = [[XJYBarItem alloc] initWithDataNumber:@(-7.04) color:XJYDarkBlue dataDescribe:@"test"];
        [itemArray addObject:item12];
        XJYBarItem *item13 = [[XJYBarItem alloc] initWithDataNumber:@(44.99) color:XJYDarkBlue  dataDescribe:@"test"];
        [itemArray addObject:item13];
        XJYBarItem *item14 = [[XJYBarItem alloc] initWithDataNumber:@(28.48) color:XJYDarkBlue dataDescribe:@"test"];
        [itemArray addObject:item14];
        XJYBarItem *item15 = [[XJYBarItem alloc] initWithDataNumber:@(-52.91) color:XJYDarkBlue dataDescribe:@"test"];
        [itemArray addObject:item15];
        
        XJYBarItem *item16 = [[XJYBarItem alloc] initWithDataNumber:@(0.93) color:XJYDarkBlue dataDescribe:@"test"];
        [itemArray addObject:item16];
        XJYBarItem *item17 = [[XJYBarItem alloc] initWithDataNumber:@(77.04) color:XJYDarkBlue dataDescribe:@"test"];
        [itemArray addObject:item17];
        XJYBarItem *item18 = [[XJYBarItem alloc] initWithDataNumber:@(4.99) color:XJYDarkBlue  dataDescribe:@"test"];
        [itemArray addObject:item18];
        XJYBarItem *item19 = [[XJYBarItem alloc] initWithDataNumber:@(-28.48) color:XJYDarkBlue dataDescribe:@"test"];
        [itemArray addObject:item19];
        XJYBarItem *item20 = [[XJYBarItem alloc] initWithDataNumber:@(52.91) color:XJYDarkBlue dataDescribe:@"test"];
        [itemArray addObject:item20];
        
        XJYBarItem *item21 = [[XJYBarItem alloc] initWithDataNumber:@(0.93) color:XJYDarkBlue dataDescribe:@"test"];
        [itemArray addObject:item21];
        XJYBarItem *item22 = [[XJYBarItem alloc] initWithDataNumber:@(7.04) color:XJYDarkBlue dataDescribe:@"test"];
        [itemArray addObject:item22];
        XJYBarItem *item23 = [[XJYBarItem alloc] initWithDataNumber:@(4.99) color:XJYDarkBlue  dataDescribe:@"test"];
        [itemArray addObject:item23];
        XJYBarItem *item24 = [[XJYBarItem alloc] initWithDataNumber:@(44.48) color:XJYDarkBlue dataDescribe:@"test"];
        [itemArray addObject:item24];
        XJYBarItem *item25 = [[XJYBarItem alloc] initWithDataNumber:@(52.91) color:XJYDarkBlue dataDescribe:@"test"];
        [itemArray addObject:item25];
        
        XXPositiveNegativeBarChart *barChart = [[XXPositiveNegativeBarChart alloc] initWithFrame:CGRectMake(0, 0, 375, 200) dataItemArray:itemArray topNumber:@100 bottomNumber:@(-170)];
        
        [self.contentView addSubview:barChart];
        self.selectionStyle = UITableViewCellSelectionStyleNone;
    }
    return self;
}
@end

File: XJYChartDemo/XJYChartDemo-Swift/View/PositiveNegativeBarChartCell.swift
//
//  PositiveNegativeBarChartCell.swift
//  XJYChartDemo-Swift
//
//  Created by Kelly Roach on 8/17/18.
//  Copyright © 2018 JunyiXie. All rights reserved.
//

import UIKit
import XJYChart

class PositiveNegativeBarChartCell: UITableViewCell, XJYChartDelegate {
    override func awakeFromNib() {
        super.awakeFromNib()
        // Initialization code
    }

    override func setSelected(_ selected: Bool, animated: Bool) {
        super.setSelected(selected, animated: animated)
        // Configure the view for the selected state
    }

    override init(style: UITableViewCellStyle, reuseIdentifier: String?) {
        super.init(style: style, reuseIdentifier: reuseIdentifier)

        var itemArray: [AnyHashable] = []
        let item1 = XBarItem(dataNumber: -80.93, color: Constants.XJYDarkBlue, dataDescribe: "test")
        itemArray.append(item1!)
        let item2 = XBarItem(dataNumber: -107.04, color: Constants.XJYDarkBlue, dataDescribe: "test")
        itemArray.append(item2!)
        let item3 = XBarItem(dataNumber: 77.99, color: Constants.XJYDarkBlue, dataDescribe: "test")
        itemArray.append(item3!)
        let item4 = XBarItem(dataNumber: 57.48, color: Constants.XJYDarkBlue, dataDescribe: "test")
        itemArray.append(item4!)
        let item5 = XBarItem(dataNumber: -89.91, color: Constants.XJYDarkBlue, dataDescribe: "test")
        itemArray.append(item5!)
        let item6 = XBarItem(dataNumber: 66.93, color: Constants.XJYDarkBlue, dataDescribe: "test")
        itemArray.append(item6!)
        let item7 = XBarItem(dataNumber: 7.04, color: Constants.XJYDarkBlue, dataDescribe: "test")
        itemArray.append(item7!)
        let item8 = XBarItem(dataNumber: -77.99, color: Constants.XJYDarkBlue, dataDescribe: "test")
        itemArray.append(item8!)
        let item9 = XBarItem(dataNumber: -28.48, color: Constants.XJYDarkBlue, dataDescribe: "test")
        itemArray.append(item9!)
        let item10 = XBarItem(dataNumber: 52.91, color: Constants.XJYDarkBlue, dataDescribe: "test")
        itemArray.append(item10!)
            //
        let item11 = XBarItem(dataNumber: -0.93, color: Constants.XJYDarkBlue, dataDescribe: "test")
        itemArray.append(item11!)
        let item12 = XBarItem(dataNumber: -7.04, color: Constants.XJYDarkBlue, dataDescribe: "test")
        itemArray.append(item12!)
        let item13 = XBarItem(dataNumber: 44.99, color: Constants.XJYDarkBlue, dataDescribe: "test")
        itemArray.append(item13!)
        let item14 = XBarItem(dataNumber: 28.48, color: Constants.XJYDarkBlue, dataDescribe: "test")
        itemArray.append(item14!)
        let item15 = XBarItem(dataNumber: -52.91, color: Constants.XJYDarkBlue, dataDescribe: "test")
        itemArray.append(item15!)
        let item16 = XBarItem(dataNumber: 0.93, color: Constants.XJYDarkBlue, dataDescribe: "test")
        itemArray.append(item16!)
        let item17 = XBarItem(dataNumber: 77.04, color: Constants.XJYDarkBlue, dataDescribe: "test")
        itemArray.append(item17!)
        let item18 = XBarItem(dataNumber: 4.99, color: Constants.XJYDarkBlue, dataDescribe: "test")
        itemArray.append(item18!)
        let item19 = XBarItem(dataNumber: -28.48, color: Constants.XJYDarkBlue, dataDescribe: "test")
        itemArray.append(item19!)
        let item20 = XBarItem(dataNumber: 52.91, color: Constants.XJYDarkBlue, dataDescribe: "test")
        itemArray.append(item20!)
        let item21 = XBarItem(dataNumber: 0.93, color: Constants.XJYDarkBlue, dataDescribe: "test")
        itemArray.append(item21!)
        let item22 = XBarItem(dataNumber: 7.04, color: Constants.XJYDarkBlue, dataDescribe: "test")
        itemArray.append(item22!)
        let item23 = XBarItem(dataNumber: 4.99, color: Constants.XJYDarkBlue, dataDescribe: "test")
        itemArray.append(item23!)
        let item24 = XBarItem(dataNumber: 44.48, color: Constants.XJYDarkBlue, dataDescribe: "test")
        itemArray.append(item24!)
        let item25 = XBarItem(dataNumber: 52.91, color: Constants.XJYDarkBlue, dataDescribe: "test")
        itemArray.append(item25!)
        let barChart = XPositiveNegativeBarChart(frame: CGRect(x: 0, y: 0, width: 375, height: 200), dataItemArray: NSMutableArray(array: itemArray), topNumber: 100, bottomNumber: -170)
        barChart!.barChartDelegate = self
        contentView.addSubview(barChart!)
        selectionStyle = UITableViewCellSelectionStyle.none
        
    }
    
    required init?(coder aDecoder: NSCoder) {
        fatalError("init(coder:) has not been implemented")
    }
    
    func userClickedOnBar(at idx: Int) {
        print("XBarChartDelegate touch Bat At idx \(idx)")
    }
}
